packages/@uppy/image-generator/src/index.tsx

```diff
@@ -0,0 +1,288 @@
+import type {
+  Body,
+  Meta,
+  MinimalRequiredUppyFile,
+  UIPluginOptions,
+  UppyEventMap,
+} from '@uppy/core'
+import Uppy, { UIPlugin } from '@uppy/core'
+import { FilterInput, SearchView } from '@uppy/provider-views'
+import type { AssemblyOptions, AssemblyResult } from '@uppy/transloadit'
+import Transloadit from '@uppy/transloadit'
+import locale from './locale.js'
+
+export interface ImageGeneratorOptions extends UIPluginOptions {
+  assemblyOptions: (prompt: string) => Promise<AssemblyOptions>
+}
+
+interface PluginState extends Record<string, unknown> {
+  prompt: string
+  results: AssemblyResult[]
+  checkedResultIds: Set<AssemblyResult['id']>
+  loading: boolean
+  loadingMessageIndex: number
+  firstRun: boolean
+}
+
+const defaultState = {
+  prompt: '',
+  results: [],
+  checkedResultIds: new Set(),
+  loading: false,
+  loadingMessageIndex: 0,
+  firstRun: true,
+} satisfies PluginState
+
+const LOADING_MESSAGES = [
+  'generating1',
+  'generating2',
+  'generating3',
+  'generating4',
+  'generating5',
+] as const
+
+export default class ImageGenerator<
+  M extends Meta,
+  B extends Body,
+> extends UIPlugin<ImageGeneratorOptions, M, B, PluginState> {
+  private loadingInterval: ReturnType<typeof setInterval> | null = null
+
+  constructor(uppy: Uppy<M, B>, opts: ImageGeneratorOptions) {
+    super(uppy, opts)
+
+    this.id = this.opts.id || 'ImageGenerator'
+    this.title = 'Image Generator'
+    this.type = 'acquirer'
+
+    this.defaultLocale = locale
+
+    this.setPluginState(defaultState)
+
+    this.i18nInit()
+  }
+
+  install(): void {
+    const { target } = this.opts
+    if (target) {
+      this.mount(target, this)
+    }
+  }
+
+  uninstall(): void {
+    this.clearLoadingInterval()
+    this.unmount()
+  }
+
+  private clearLoadingInterval(): void {
+    if (this.loadingInterval) {
+      clearInterval(this.loadingInterval)
+      this.loadingInterval = null
+    }
+  }
+
+  private startLoadingAnimation(): void {
+    this.clearLoadingInterval()
+    this.loadingInterval = setInterval(() => {
+      const { loadingMessageIndex } = this.getPluginState()
+      const nextIndex = (loadingMessageIndex + 1) % LOADING_MESSAGES.length
+      this.setPluginState({ loadingMessageIndex: nextIndex })
+    }, 4000)
+  }
+
+  search = async () => {
+    const assemblyOptions = await this.opts.assemblyOptions(
```

Comment:
Here we're being hit by the banana/gorilla problem because our code is very OOP.

> You wanted a banana but what you got was a gorilla holding the banana and the entire jungle.

the banana is wanting to execute an assembly, the gorilla is the Transloadit plugin with all its events and quirks, and the Uppy class is the entire jungle. It doesn't have anything to do with the problem at hand but we still need to instantiate to get the banana.

```suggestion
    // This banana/gorilla problem is because of our assembly code is very OOP.
    // TODO rewrite the Transloadit plugin so that all its assembly creation, execution and response is abstracted out into a simple to use function that can be used by the Transloadit plugin as well as other consumers (like here)
    const assemblyOptions = await this.opts.assemblyOptions(
```